osxkeychain.go

```diff
@@ -34,6 +34,7 @@ type GenericPasswordAttributes struct {
 	AccountName         string
 	Password            []byte
 	TrustedApplications []string
+	Keychain            []string
```

Comment:
This is problematic, since it's used in one way for AddGenericPassword, but used in another for FindGenericPassword. Hmm. The general problem is that GenericPasswordAttributes really needs to be split up into two structs: one for specifying the attributes of a generic password to add, and another to specify search parameters for FindGenericPassword.

Can you try doing that? Feel free to make a separate pull request for it.